### ISSUE TITLE
The padding embeddings assertion will fail during training because the embedding layer is initialized with random values, not zeros. The padding_idx parameter only prevents gradient updates to padding embeddings during backpropagation, but doesn't initialize them to zero. This test should either check that padding gradients are not updated, or initialize the embedding layer with zeros for padding indices before testing.

### ISSUE THREAD

**Issue by tzervas** [tzervas/tritter#25]
The padding embeddings assertion will fail during training because the embedding layer is initialized with random values, not zeros. The padding_idx parameter only prevents gradient updates to padding embeddings during backpropagation, but doesn't initialize them to zero. This test should either check that padding gradients are not updated, or initialize the embedding layer with zeros for padding indices before testing.

_Originally posted by @copilot-pull-request-reviewer in https://github.com/tzervas/tritter/pull/2#discussion_r2713634153_
